### DISCUSSION TITLE
Adicionar referência cruzada

### DISCUSSION THREAD

**Discussion by george-gca** [levxyca/diciotech#149] [Ideas]
Muitos conceitos acabam por citar outros conceitos internamente. Por exemplo, a definição atual do C++:

```
C++ é uma linguagem de programação compilada multi-paradigma
(seu suporte inclui linguagem imperativa, orientada a objetos
e genérica) e de uso geral...
```

Seria legal se, ao citar outros conceitos, ele tivesse um hiperlink para a definição desses outros conceitos, em uma parte como `Conceitos relacionados`. Dessa forma seria possível somente com o mouse por exemplo sair navegando entre as definições necessárias para entender o conceito principal.

**Comment by levxyca** [comment]
Muito legal sua sugestão @george-gca! Dependendo do contexto, pode ser interessante mesmo. Um ponto que me foi levantado sobre isso, é que pode ser algo que quebre um pouco a experiência da pessoa usuária, por exemplo, quando usamos o dicionário para buscar uma palavra e ele tem o significado de outra palavra que a gente tem que pesquisar de novo.... isso pode ser bem frustrante.

Acho que é legal implementarmos isso, pensando sempre se realmente existe a necessidade ou se podemos apenas simplificar mais a explicação.

**Reply by george-gca** [reply]
Eu acho que poderiam ser links como a wikipedia faz, daí clicando na palavra que faça referência cruzada leve a outra busca na mesma janela mesmo, o que permitiria usar o voltar do navegador pra voltar pra palavra anterior, mas como atualmente a busca é feita a cada letra digitada não sei como seria implementado de fato.

Pensei nisso porque seria útil tanto do ponto de vista de navegação (e até acessibilidade), quanto pra entender o conceito mesmo. O exemplo ali em cima do c++ é um bom exemplo, envolve muitos outros conceitos na definição que usaram. Imagina nesse caso, pra cada termo novo (linguagem de programação compilada, linguagem imperativa, orientada a objetos, etc) a pessoa usuária tem que fazer uma nova busca por esse termo e digitar ele. Às vezes o termo pode ser até uma palavra meio difícil de digitar ou lembrar, então sendo um link direto seria útil.
